src/Radicle/Internal/Effects.hs

```diff
@@ -188,7 +188,7 @@ replPrimFns = fromList $ allDocs $
               [ (Keyword (Ident "private-key"), skv)
               , (Keyword (Ident "public-key"), pkv)
               ]
-          xs -> throwErrorHere $ WrongNumberOfArgs "gen-key-pair!" 0 (length xs)
+          xs -> throwErrorHere $ WrongNumberOfArgs "gen-key-pair!" 1 (length xs)
```

Comment:
Maybe use `oneArg` instead?

Reply:
@jameshaydon how is this even working? It doesn't seem to eval it's arguments at all, but the `(default-ecc-curve)` function definitely gets called. 

(Speaking of which, why is `(default-ecc-curve)` a function in the first place?)

Reply:
@jkarni 
> It doesn't seem to eval it's arguments at all

what? All functions eval their arguments.

Yes, I will add it to the pure env directly. C.f. #209